app/models/challenge_assignment.rb

```diff
@@ -53,6 +53,15 @@ def signups_match
 
   scope :order_by_offering_pseud, -> { joins(OFFERING_PSEUD_JOIN).order("pseuds.name") }
 
+  scope :offer_signups_by, lambda { |user|
+    joins(offer_signup: { pseud: :user })
+      .where(users: { id: user.id })
+  }
+
+  scope :pinch_hits_by, lambda { |user|
+    joins(pinch_hitter: :user)
+      .where(users: { id: user.id })
+  }
```

Comment:
Both of these scopes contain more joins than necessary, which may impact the efficiency. Was there something wrong with the previous pair of queries that you were using? https://github.com/otwcode/otwarchive/blob/b79c75b7e4859148885a745f63cd9762a3f9cbad/app/helpers/works_helper.rb#L198

Reply:
🥹 I was very out of my element when I returned to look at the code for challenges.  Pardon my confusion.